interface/patient_file/deleter.php

```diff
@@ -32,7 +32,7 @@
 require_once($GLOBALS['srcdir'].'/sl_eob.inc.php');
 
  $patient     = $_REQUEST['patient'];
- $encounterid = $_REQUEST['encounterid'];
+ $encounterid = filter_var($_REQUEST['encounterid'], FILTER_VALIDATE_INT);
```

Comment:
What vulnerability are you fixing in this script. $encounterid is properly escaped already in this entire script.?
Better to avoid doing things that are not needed. Sometimes int filters like this have size limits; for example, what happens if we migrate  encounterid to use uid in the future.

Reply:
Because  $encounterid is printed into js (line 412) it's possible to inject script without <script> tags and filter_var prevents this..
![image](https://user-images.githubusercontent.com/17809866/30794147-37ba5086-a1cf-11e7-8c51-e0d455e136fa.png)

Reply:
But doesn't the attr($encounterid) kill that vulnerability?
(note the hardening of this script was recent and a good chance the vulnerability report you got was before this script was hardened with the attr() etc.)

Reply:
Above not yet addressed.

Reply:
I attached incorrect issue. I also don't understand how they inject <script> tags although attr function (they tested code from last months ). but as I wrote in possible to inject without <script> tag because it's in the middle of script like -
![image](https://user-images.githubusercontent.com/17809866/30898597-c96113e4-a364-11e7-8413-dcd012a84b5e.png)
I succeed to reproduce with this parameter - deleter.php?encounterid=9);alert(123);//

Reply:
Very interesting and sounds like a potentially global vulnerability. Would be better to find a solution that does not depend on the integer, since there are many cases where this vulnerability will come up and won't be able to use the integer method.
How about the following method(which appears to escape parentheses thus may shut this down):
urlencode()
And if that doesn't work, then may need to try(this would be best to avoid if possible since it can require hard coding of the encoding):
htmlentities()

Reply:
Also, to confirm we are working on the right place, is the the vulnerable spot in the code?
``````
echo "window.opener.imdeleted(" . attr($encounterid) . ");\n";
``````

Reply:
Also should try addslashes() (albeit I doubt that will work).

Reply:
@sunsetsystems ,
Any thoughts on how to escape the line of code above to prevent this vulnerability?

Reply:
How about this(placed single qoutes outside the double quotes):
``````
echo "window.opener.imdeleted('" . attr($encounterid) . "');\n";
``````

Reply:
As a sanity check, I am not able to confirm this vulnerability. I placed the path you gave above and no alert popup happens. here's the screenshot:
![image](https://user-images.githubusercontent.com/278968/30901246-d6c1f1b4-a31b-11e7-8c9a-374355a7a4d9.png)